search/includes/classes/class-search.php

```diff
@@ -484,6 +485,12 @@ protected function setup_hooks() {
 
 		add_filter( 'vip_search_post_meta_allow_list', array( $this, 'filter__vip_search_post_meta_allow_list_defaults' ) );
 
+		// Limit the max result window
+		add_filter( 'ep_max_result_window', [ $this, 'limit_max_result_window' ], PHP_INT_MAX );
```

Comment:
A quick inline comment that indicates which one of these is for the index settings and which one is for the query parameters would be helpful for our future selves.

Reply:
Nice idea!